src/Tribe/Modules/Maintenance.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace Tribe\Extensions\EA_Additional_Options\Modules;
+
+class Maintenance {
+	public function hook() {
+		add_action( 'init', [ $this, 'cron_setup' ] );
+		add_action( 'tec_delete_old_ignored_events_cron', [ $this, 'delete_ignored_events' ] );
+	}
+
+	/**
+	 * Set up a cron job to delete old ignored events.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @return void
+	 */
+	public function cron_setup() {
+		$range        = tribe_get_option( Settings::PREFIX . 'ignore_range', null );
+		$schedule     = tribe_get_option( Settings::PREFIX . 'ignore_schedule', 'daily' );
+		$set_schedule = wp_get_schedule( 'tec_delete_old_ignored_events_cron' );
+
+		// Delete the cron if setting is disabled or the schedule changed, and it's not manually run.
+		if (
+			empty( $range )
+			|| $schedule !== $set_schedule
+			   && $set_schedule
+		) {

```

Comment:
No need to check if `$set_schedule` is false. Simply comparing to  `$schedule` is enough.
```suggestion
		if ( empty( $range ) || $schedule !== $set_schedule ) {
```

Reply:
Need to check in case it's a manually run cron with WP Crontrol.